csrc/scheduler/reduction.cpp

```diff
@@ -1210,12 +1210,15 @@ void scheduleReduction(Fusion* fusion, const ReductionParams& rparams) {
   const bool vectorize =
       rparams.vectorize_inner_reduction || rparams.vectorize_iter_dom;
 
-  // allow iter domain grouped reduction for block outer reduction.
+  // allow iter domain grouped reduction for block and grid outer reductions.
   // TODO: the var name is confusing, should rename
   // [cross_grid/block_inner_reduction] to [cross_grid/block_reduction], see
   // https://github.com/NVIDIA/Fuser/issues/1863
-  const bool use_iter_grouped_reduction = !rparams.cross_grid_inner_reduction &&
-      !rparams.fastest_dim && rparams.cross_block_inner_reduction;
+  const bool has_welford = ir_utils::hasOpsOfType<WelfordOp>(fusion);
```

Comment:
`welford` is a special reduction, grouped welford is only enabled for grid persistent.
Before this PR, we allow block reduce to `use_iter_grouped_reduction`, it doesn't cause error because we don't have a vectorized block welford outer reduction in our tests.
In this PR we want to extend to `grid reduce`, so needs to do special check for welford.

Reply:
Thanks, I understand it as this is an intended behavior change.